application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/apiuser/ApiUserService.java

```
@@ -0,0 +1,87 @@
+import ch.admin.seco.jobs.services.jobadservice.core.domain.events.DomainEventPublisher;
+import ch.admin.seco.jobs.services.jobadservice.core.time.TimeMachine;
+import ch.admin.seco.jobs.services.jobadservice.domain.apiuser.ApiUser;
+import ch.admin.seco.jobs.services.jobadservice.domain.apiuser.ApiUserId;
+import ch.admin.seco.jobs.services.jobadservice.domain.apiuser.ApiUserRepository;
+import ch.admin.seco.jobs.services.jobadservice.domain.apiuser.events.ApiUserSavedEvent;
+
+@Service
+@Transactional
+public class ApiUserService {
+
+	private ApiUserRepository apiUserRepository;
+
+	@Autowired
+	public ApiUserService(ApiUserRepository apiUserRepository) {
+		this.apiUserRepository = apiUserRepository;
+	}
+
+	public ApiUserId createApiUser(CreateApiUserDto createApiUserDto) {
+		ApiUser.Builder builder = new ApiUser.Builder();
+		builder.setId(new ApiUserId());
+		builder.setUsername(createApiUserDto.getUsername());
+		builder.setPassword(createApiUserDto.getPassword());
+		builder.setEmail(createApiUserDto.getEmail());
+		builder.setActive(createApiUserDto.getActive());
+		builder.setCompanyName(createApiUserDto.getCompanyName());
+		builder.setContactName(createApiUserDto.getContactName());
+		builder.setContactEmail(createApiUserDto.getContactEmail());
+		builder.setCreateDate(TimeMachine.now().toLocalDate());
+
+		ApiUser apiUser = apiUserRepository.save(new ApiUser(builder));
+		DomainEventPublisher.publish(new ApiUserSavedEvent(apiUser));
+		return apiUser.getId();
+	}
+
+	public void changeApiUserStatus(ApiUserId id, boolean activeStatus) {
+		ApiUser apiUser = getById(id);
+		apiUser.setActive(activeStatus);
+		DomainEventPublisher.publish(new ApiUserSavedEvent(apiUser));
+	}
+
+	private ApiUser getById(ApiUserId id) {
+		return apiUserRepository.findById(id)
+				.orElseThrow(() -> new AggregateNotFoundException(ApiUser.class, id.getValue()));
+	}
+
+	public ApiUserDto findById(ApiUserId apiUserId) {
```

Per definition should a JPA-get always throw an exception if not found, but a JPA-find should return null